### PR TITLE
bpo-30923: Silence fall-through warnings included in -Wextra since gc…

### PR DESCRIPTION
**Pull request by skrah**
…c-7.0.

https://bugs.python.org/issue30923

### CODE REVIEW

File: Modules/_ctypes/_ctypes.c [hunk 1]
```diff
@@ -3614,7 +3614,7 @@ _build_callargs(PyCFuncPtrObject *self, PyObject *argtypes,
         case (PARAMFLAG_FIN | PARAMFLAG_FOUT):
             *pinoutmask |= (1 << i); /* mark as inout arg */
             (*pnumretvals)++;
-            /* fall through to PARAMFLAG_FIN... */
+            /* fall through */
```
Comment (serhiy-storchaka) [inline Modules/_ctypes/_ctypes.c:3617]: Why existing comments are shortened?

Comment (skrah) [reply Modules/_ctypes/_ctypes.c:3617]: gcc only recognizes "fall through" (and a couple of other versions), but not that comment with additional words.

I only changed comments that aren't recognized.

File: Python/compile.c [hunk 2]
```diff
@@ -4154,6 +4154,7 @@ expr_constant(struct compiler *c, expr_ty e)
         else if (o == Py_False)
             return 0;
     }
+    /* fall through */
```
Comment (serhiy-storchaka) [inline Python/compile.c:4157]: Wouldn't be better to move this comment one line up, inside a block, and align it with `if` and `else`? And add "else" before "fall through".

Comment (skrah) [reply Python/compile.c:4157]: Agreed, but gcc doesn't recognize that.  It's pretty inflexible.

**Review by serhiy-storchaka — APPROVED** [review]

**Review by tiran — APPROVED** [review]
LGTM, thanks!

**Comment by skrah** [comment]
Thanks everyone!

**Comment by skrah** [comment]
Just checking out the bot, this is fantastic.

**Comment by Mariatta** [comment]
Thanks @skrah Seems like there was problem backporting. I will try this again.

**Comment by skrah** [comment]
Okay, thanks!  I suspected there would be too many conflicts. So let's not backport this, it isn't worth the trouble.

**Pull request merged**